### PR TITLE
Serve API using Compojure-API, thus enabling Swagger docs

### PR DESCRIPTION
**Pull request by mforsyth**

### CODE REVIEW

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -403,21 +452,16 @@
...
-                              [false {::jobs (mapv #(UUID/fromString %) jobs)}]
-                              (catch Exception e
-                                [true {::error e}])))
-                          [true {::error "must supply at least one job query param"}])
+  (-> (liberator/resource
+       :available-media-types ["application/json"]
+       :allowed-methods [:post :get :delete]
+       :malformed? (fn [ctx]
+                     (condp contains? (get-in ctx [:request :request-method])
+                       ;; schema will already validate.
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/api.clj]: Why will it already validate?

**Comment by wyegelwel** [comment]
This LGTM! I'm really excited about this.

File: scheduler/src/cook/mesos/api.clj (hunk 1 above)
Comment (mforsyth) [inline scheduler/src/cook/mesos/api.clj]: Before the request arrives at the "job-resource" handler, the request body will have been coerced to a specific schema.  Coercing to that schema already does everything that the inline validation here used to do (prove that the job parameters are an array of UUID's with at least 1 entry), and provides better error messages to boot.

Comment (mforsyth) [inline scheduler/src/cook/mesos/api.clj]: (Basically, if the coercion fails, you'll get a 400 response with a JSON structure describing why the coercion failed).

**Pull request merged**